core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -372,5 +372,5 @@ private[spark] object SparkConf {
   /**
    * Return whether the given config is a Spark port config.
    */
-  def isSparkPortConf(name: String): Boolean = name.startsWith("spark.") && name.endsWith(".port")
+  def isSparkPortConf(name: String): Boolean = name.startsWith("spark.") && name.contains(".port")
```

Comment:
I understand that, but this currently matches something like `spark.portable.mushroom` which has nothing to do with Spark ports. Maybe instead you want to do something like:
```
name.matches("spark\..*\.port") | name.startsWith("spark.port.")
```